src/Components/MomentReacter.tsx

```diff
@@ -0,0 +1,120 @@
+import React, { useEffect, useState, useRef } from 'react';
+import moment from 'moment';
+import _ from 'lodash';
+
+const FILE_LOG_TAG = 'MomentReacter';
+const MAX_TIMEOUT_TIME = 2147483647;
+interface NextMomentResult {
+    theMoment: moment.Moment;
+    theDiff: number;
+}
+
+const nextMoment = (referenceMoment: moment.Moment, moments: Array<moment.Moment>): NextMomentResult | null => {
+    let result: NextMomentResult | null = null;
+    moments.forEach(theMoment => {
```

Comment:
Might've been cleaner with a reduce, but nice.

Reply:
True, I'm not going to change it to a reduce now but I might do that in the future